File: merge-two-sorted-lists/merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        s1, s2 = l1, l2
        answer = ListNode(None)
        node = answer
        
        while s1 and s2:
            if s1.val < s2.val:
                node.next = ListNode(s1.val)
                node = node.next
                s1 = s1.next
            else:
                node.next = ListNode(s2.val)
                node = node.next
                s2 = s2.next
        
        if s1:
            node.next = s1
        if s2:
            node.next = s2
        
        return answer.next

File: merge-two-sorted-lists/merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode root = new ListNode();
        ListNode result = root;

        while (l1 != null && l2 != null) {
            if (l1.val < l2.val) {
                result.next = new ListNode(l1.val);
                result = result.next;
                l1 = l1.next;
            }
            else {
                result.next = new ListNode(l2.val);
                result = result.next;
                l2 = l2.next;
            }
        }
        result.next = (l1 == null ? l2 : l1);

        return root.next;
    }
}